diff --git a/gfx/drivers/gl2.c b/gfx/drivers/gl2.c
--- a/gfx/drivers/gl2.c
+++ b/gfx/drivers/gl2.c
@@ -1088,6 +1088,13 @@ static void gl2_raster_font_render_msg(
    else
       gl2_raster_font_setup_viewport(gl, font, width, height, full_screen,
             config_get_ptr()->bools.video_scale_integer);
+
+   if (gl2_menu_display_rotation())
+   {
+      x *= (float)gl->vp.width / (float)gl2_menu_font_width(gl);
+      y  = 1.0f - ((1.0f - y) *
+            (float)gl->vp.height / (float)gl2_menu_font_height(gl));
+   }
 
    if (    !string_is_empty(msg)
          && font->font_data
