diff --git a/CMakeLists.txt b/CMakeLists.txt
index 910f322..74a1468 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -324,11 +324,13 @@ endif()
 
 if(NOT LIBRETRO AND NOT IOS AND NOT MACOSX)
 	find_package(SDL2)
-	find_package(SDL2_ttf)
+	if (NOT PLUMOS_PIXEL2)
+		find_package(SDL2_ttf)
+	endif()
 	find_package(Fontconfig)
 
 	# TODO: this can be removed once CI supports newer SDL2_ttf
-	if (NOT SDL2_ttf_FOUND)
+	if (NOT PLUMOS_PIXEL2 AND NOT SDL2_ttf_FOUND)
 		find_package(PkgConfig)
 		if(PkgConfig_FOUND)
 			pkg_check_modules(SDL2_ttf_PKGCONFIG IMPORTED_TARGET SDL2_ttf)
