diff --git a/libretro/cmdline.c b/libretro/cmdline.c
--- a/libretro/cmdline.c
+++ b/libretro/cmdline.c
@@ -77,12 +77,18 @@ void CreateCommandLine(void)
       Add_Option("--machine");
       Add_Option(hatari_machinetype);
       Add_Option("--memsize");
       Add_Option(hatari_ramsize);
-      Add_Option("--disk-a");
-      Add_Option(RPATH[0]==0 ? "":RPATH/*ARGUV[0]*/);
-      Add_Option("--disk-b");
-      Add_Option(hatari_autoloadb==true ? RPATH2[0]==0 ? "" : RPATH2 : "");
+      if (RPATH[0] != 0)
+      {
+          Add_Option("--disk-a");
+          Add_Option(RPATH/*ARGUV[0]*/);
+      }
+      if (hatari_autoloadb == true && RPATH2[0] != 0)
+      {
+          Add_Option("--disk-b");
+          Add_Option(RPATH2);
+      }
       Add_Option("--protect-floppy");
       Add_Option(hatari_writeprotect_floppy);
       Add_Option("--protect-hd");
       Add_Option(hatari_writeprotect_hd);
@@ -88,7 +94,10 @@ void CreateCommandLine(void)
       if (RETRO_GD[0] == 0 && RETRO_IDE[0]==0)
       {
-          Add_Option("--acsi");
-          Add_Option(RETRO_HD[0] == 0 ? "" : RETRO_HD);
+          if (RETRO_HD[0] != 0)
+          {
+              Add_Option("--acsi");
+              Add_Option(RETRO_HD);
+          }
       }
       else if (RETRO_IDE[0] == 0)
       {
